packages/playwright-core/src/server/common/componentUtils.ts

```diff
@@ -200,3 +200,13 @@ export function parseComponentSelector(selector: string): ParsedComponentSelecto
     throw new Error(`Error while parsing selector \`${selector}\` - selector cannot be empty`);
   return result;
 }
+
+export function isInsideScope(scope: Node, node: Node): boolean {
+  while (node) {
+    if (scope.contains(node))
+      return true;
+    node = node.parentNode || (node as ShadowRoot).host;
```

Comment:
There is an `HTMLAnchorElement.host` property  that breaks this code. Take a look at `parentNodeOrShadowHost` method we have somewhere.

Reply:
Done